spicy/toolchain/include/ast/types/unit.h

```diff
@@ -80,9 +80,22 @@ class Unit : public UnqualifiedType {
      */
     auto isPublic() const { return _public; };
 
-    /** * Returns true if this unit type can act as a filter. */
+    /** Returns true if this unit type can act as a filter. */
     bool isFilter() const { return propertyItem("%filter") != nullptr; }
 
+    /**
+     * Returns true if this unit may have a filter attached. This is derived
+     * from actual use of the type inside the global AST and will be valid only
+     * after the Spicy resolver has finished.
+     */
+    auto mayHaveFilter() const { return _may_have_filter; }
+
+    /**
+     * Sets the flag indicating whether the unit may see a filter getting
+     * attached. Should be called only by the resolver.
+     */
+    void setMayHaveFilter(bool may_have_filter) { _may_have_filter = may_have_filter; }
```

Comment:
Currently we never unset this (and I suspect it would be difficult since filters could be removed in a number of ways), so for now we could make this a one-way toggle, e.g.,

```suggestion
    void setMayHaveFilter() { _may_have_filter = true; }
```

Reply:
As a general pattern I like to pass booleans to functions so that one can go both ways.